File: go/hamming/hamming.go
// Package hamming implements hamming distance
package hamming

import "fmt"

const testVersion = 6

// Distance returns Hamming distance between two strings of equal length
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return -1, fmt.Errorf("Two strings of unequal length")
	}
	distance := 0
	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			distance += 1
		}
	}
	return distance, nil
}

File: java/hamming/src/main/java/Hamming.java
public class Hamming {

    private String leftStrand;
    private String rightStrand;

    Hamming(String leftStrand, String rightStrand) {
        this.leftStrand = leftStrand;
        this.rightStrand = rightStrand;
        if (this.leftStrand.length() != this.rightStrand.length()) {
            throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");
        }
    }

    int getHammingDistance() {
        int end = this.leftStrand.length();
        int distance = 0;
        for (int i = 0; i < end; i++) {
            if (this.leftStrand.charAt(i) != this.rightStrand.charAt(i)) {
                distance += 1;
            }
        }
        return distance;
    }

}
